google-cloud-clients/google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerImpl.java

```diff
@@ -905,6 +917,8 @@ TransactionContextImpl newTransaction() {
     }
     
     <T extends SessionTransaction> T setActive(@Nullable T ctx) {
+      throwIfTransactionsPending();
```

Comment:
what happens when they do this ?

txn1 = client.readWriteTransaction();
txn2 = client.readWriteTransaction(); // should it fail here ?

Reply:
No, I think it shouldn't fail here because they have not run the transaction. It's ok to have multiple prepares.